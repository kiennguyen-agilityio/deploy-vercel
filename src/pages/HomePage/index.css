.product-title {
  width: 163px;
  height: 54px;
  font: normal var(--font-weight-extrabold) var(--font-size-xl) / var(--line-height-xxl) 'Open Sans';
  color: var(--color-black);
  margin: 50px 50px 50px 20px;
}

.product-content {
  display: flex;
}
