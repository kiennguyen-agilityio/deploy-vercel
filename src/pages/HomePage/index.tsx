import { useState, lazy, Suspense, useCallback } from 'react';

// components
import Filter from '@/pages/HomePage/Filter/Filter';
import LoadingSpinner from '@/components/Common/LoadingSpinner/LoadingSpinner';

// layouts
import DefaultLayout from '@/layouts/DefaultLayout/DefaultLayout';

// style
import './index.css';

/**
 * Represents the ProductList component.
 *
 * This component is used to lazy load the ListProduct component.
 */
const ProductList = lazy(() => import('@/components/ListProduct/ListProduct'));

const HomePage: React.FC = () => {
  const [selectedPrice] = useState<number | null>(null);
  const [selectedCategory, setSelectedCategory] = useState<string>('');
  const [sortType, setSortType] = useState<'asc' | 'desc' | ''>('');
  const [selectedSubcategory, setSelectedSubcategory] = useState<string>('');
  const [selectedColors, setSelectedColors] = useState<string[]>([]);

  /**
   * Handles the change of colors.
   * @param colors - An array of color strings.
   */
  const handleColorChange = useCallback((colors: string[]): void => {
    setSelectedColors(colors);
  }, []);

  /**
   * Handles the change of category.
   * @param category - The selected category.
   */

  const handleCategoryChange = useCallback((category: string): void => {
    setSelectedCategory(category);
  }, []);

  /**
   * Handles the sorting of items by price in ascending order.
   */
  const handleSortByPriceAsc = useCallback((): void => {
    setSortType('asc');
  }, []);

  /**
   * Handles the sorting of items by price in descending order.
   */

  const handleSortByPriceDesc = useCallback((): void => {
    setSortType('desc');
  }, []);

  /**
   * Handles the change of subcategory.
   * @param subcategory - The selected subcategory.
   */
  const handleSubcategoryChange = useCallback((subcategory: string): void => {
    setSelectedSubcategory(subcategory);
  }, []);

  return (
    <DefaultLayout>
      <h2 className="product-title">Product</h2>
      <div className="product-content">
        <Filter
          selectedColors={selectedColors}
          selectedCategory={selectedCategory}
          onColorChange={handleColorChange}
          onCategoryChange={handleCategoryChange}
          onSortByPriceAsc={handleSortByPriceAsc}
          onSortByPriceDesc={handleSortByPriceDesc}
          selectedSubcategory={selectedSubcategory}
          onSubcategoryChange={handleSubcategoryChange}
        />

        {/* Apply suspense for ProductList component */}
        <Suspense fallback={<LoadingSpinner />}>
          <ProductList
            selectedColors={selectedColors}
            selectedPrice={selectedPrice}
            sortType={sortType}
            selectedSubcategory={selectedSubcategory}
          />
        </Suspense>
      </div>
    </DefaultLayout>
  );
};

export default HomePage;
