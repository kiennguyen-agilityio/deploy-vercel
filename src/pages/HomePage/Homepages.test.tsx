import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import HomePage from '.';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: (initialState: any) => [initialState, jest.fn()],
}));
describe('HomePage', () => {
  test('renders the component correctly', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
