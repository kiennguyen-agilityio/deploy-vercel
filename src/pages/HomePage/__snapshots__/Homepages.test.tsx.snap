// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="navbar"
    >
      <a
        href="/"
      >
        <div
          class="logo"
        >
          <i
            class="fa-sharp fa-solid fa-bag-shopping"
          />
          <span
            class="shop-name"
          >
            Shop Bags
          </span>
        </div>
      </a>
      <div
        class="search-bar-container"
      >
        <input
          class="search-bar"
          value=""
        />
        <i
          class="search-icon fa-sharp fa-solid fa-search"
        />
      </div>
      <div
        class="icons"
      >
        <a
          class="cart-icon-container"
          href="/cart"
        >
          <i
            class="material-symbols-outlined icon"
          >
            shopping_cart
          </i>
        </a>
        <i
          class="material-symbols-outlined icon"
        >
          favorite
        </i>
      </div>
      <div
        class="button-container"
      >
        <button
          class="btn-sign-up"
          type="button"
        >
          Sign Up
        </button>
        <button
          class="btn-login"
          type="button"
        >
          Login
        </button>
      </div>
    </div>
    <main
      class="main-content"
    >
      <h2
        class="product-title"
      >
        Product
      </h2>
      <div
        class="product-content"
      >
        <div
          class="filter-container"
        >
          <h2
            class="filter-header"
          >
            Filter
          </h2>
          <div
            class="filter-section"
          >
            <h3
              aria-hidden="true"
              class="filter-title "
              data-testid="id-color"
            >
              Color
              <span
                class="material-symbols-outlined "
              >
                expand_more
              </span>
            </h3>
          </div>
          <div
            class="filter-rating"
          >
            <h3
              aria-hidden="true"
              class="filter-title "
              data-testid="id-rating"
            >
              Rating
              <span
                class="material-symbols-outlined "
              >
                expand_more
              </span>
            </h3>
          </div>
          <div
            class="filter-price"
          >
            <h3
              aria-hidden="true"
              class="filter-title "
              data-testid="id-price"
            >
              Price
              <span
                class="material-symbols-outlined "
              >
                expand_more
              </span>
            </h3>
          </div>
          <div
            class="filter-section"
          >
            <h3
              aria-hidden="true"
              class="filter-title "
              data-testid="id-category"
            >
              Category
              <span
                class="material-symbols-outlined "
              >
                expand_more
              </span>
            </h3>
          </div>
        </div>
        <div
          class="spinner-container"
        >
          <div
            class="loading-spinner"
          />
        </div>
      </div>
    </main>
    <footer
      class="footer-distributed"
    >
      <div
        class="logo-footer"
      >
        <i
          class="fa-sharp fa-solid fa-bag-shopping"
        />
        <span
          class="shop-name"
        >
          Shop Bags
        </span>
      </div>
      <div
        class="social-icons"
      >
        <a
          href="https://www.facebook.com"
        >
          <i
            class="fa-brands fa-square-facebook"
          />
        </a>
        <a
          href="https://www.twitter.com"
        >
          <i
            class="fa-brands fa-square-twitter"
          />
        </a>
        <a
          href="https://www.linkedin.com"
        >
          <i
            class="fa-brands fa-linkedin"
          />
        </a>
        <a
          href="https://www.telegram.org"
        >
          <i
            class="fa-brands fa-square-instagram"
          />
        </a>
      </div>
      <p
        class="copyright"
      >
        2019 Â© Shopbag - All right reserved
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
